internal/resources/cluster/tkgaws/resource_tkg_aws.go

```diff
@@ -615,19 +619,30 @@ func expandTKGAWSTopology(data []interface{}) (topology *tkgawsmodel.VmwareTanzu
 		return topology
 	}
 
-	lookUpTopology, _ := data[0].(map[string]interface{})
 	topology = &tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsTopology{}
 
+	lookUpTopology, ok := data[0].(map[string]interface{})
+	if !ok {
+		if os.Getenv(helper.TmcMode) == helper.DEV {
+			log.Fatalf("[ERROR]: Topology not set: %v", data[0])
+		}
+
+		return topology
+	}
+
 	if v, ok := lookUpTopology[controlPlaneKey]; ok {
 		if v1, ok := v.([]interface{}); ok {
 			topology.ControlPlane = expandTKGAWSTopologyControlPlane(v1)
 		}
 	}
 
 	if v, ok := lookUpTopology[nodePoolsKey]; ok {
-		nodepools, _ := v.([]interface{})
-		for _, np := range nodepools {
-			topology.NodePools = append(topology.NodePools, expandTKGAWSTopologyNodePool(np))
+		if vs, ok := v.([]interface{}); ok {
```

Comment:
what is vs ? and this change is not needed.
here are not accessing node pools directly and range function does not panic so we are good here